x/gravity/module.go

```diff
@@ -0,0 +1,189 @@
+	"github.com/spf13/cobra"
+	"github.com/umee-network/umee/x/gravity/simulation"
+
+	"github.com/Gravity-Bridge/Gravity-Bridge/module/x/gravity/client/cli"
+	"github.com/Gravity-Bridge/Gravity-Bridge/module/x/gravity/keeper"
+	"github.com/Gravity-Bridge/Gravity-Bridge/module/x/gravity/types"
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/codec"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	simtypes "github.com/cosmos/cosmos-sdk/types/simulation"
+	bankkeeper "github.com/cosmos/cosmos-sdk/x/bank/keeper"
+	distrtypes "github.com/cosmos/cosmos-sdk/x/distribution/types"
+	stakingkeeper "github.com/cosmos/cosmos-sdk/x/staking/keeper"
+	abci "github.com/tendermint/tendermint/abci/types"
+)
+
+// type check to ensure the interface is properly implemented
+var (
+	_ module.AppModuleSimulation = AppModule{}
+)
+
+// AppModuleBasic object for module implementation
+type AppModuleBasic struct{}
+
+// Name implements app module basic
+func (AppModuleBasic) Name() string {
+	return types.ModuleName
+}
+
+// RegisterLegacyAminoCodec implements app module basic
+func (AppModuleBasic) RegisterLegacyAminoCodec(cdc *codec.LegacyAmino) {}
+
+// DefaultGenesis implements app module basic
+func (AppModuleBasic) DefaultGenesis(cdc codec.JSONCodec) json.RawMessage {
+	return cdc.MustMarshalJSON(types.DefaultGenesisState())
+}
+
+// ValidateGenesis implements app module basic
+func (AppModuleBasic) ValidateGenesis(cdc codec.JSONCodec, _ client.TxEncodingConfig, bz json.RawMessage) error {
+	var data types.GenesisState
+	if err := cdc.UnmarshalJSON(bz, &data); err != nil {
+		return fmt.Errorf("failed to unmarshal %s genesis state: %w", types.ModuleName, err)
+	}
+
+	return data.ValidateBasic()
+}
+
+// RegisterRESTRoutes implements app module basic
+func (AppModuleBasic) RegisterRESTRoutes(ctx client.Context, rtr *mux.Router) {}
+
+// GetQueryCmd implements app module basic
+func (AppModuleBasic) GetQueryCmd() *cobra.Command {
+	return cli.GetQueryCmd()
+}
+
+// GetTxCmd implements app module basic
+func (AppModuleBasic) GetTxCmd() *cobra.Command {
+	return cli.GetTxCmd(types.StoreKey)
+}
+
+// RegisterGRPCGatewayRoutes registers the gRPC Gateway routes for the distribution module.
+// also implements app modeul basic
+func (AppModuleBasic) RegisterGRPCGatewayRoutes(clientCtx client.Context, mux *runtime.ServeMux) {}
+
+// RegisterInterfaces implements app module basic
+func (b AppModuleBasic) RegisterInterfaces(registry codectypes.InterfaceRegistry) {
+	types.RegisterInterfaces(registry)
+}
+
+// AppModule object for module implementation
+type AppModule struct {
+	AppModuleBasic
+	keeper        keeper.Keeper
+	bankKeeper    bankkeeper.Keeper
+	accountKeeper distrtypes.AccountKeeper
+	stakingKeeper stakingkeeper.Keeper
+	cdc           codec.Codec
+}
+
+// NewAppModule creates a new AppModule Object
+func NewAppModule(
```

Comment:
Same for `AppModule`. We don't need to redefine it. Just embed the original and override any method(s) we need to.